From 23c3f2132885586b730484433183efcbd99a79a4 Mon Sep 17 00:00:00 2001
From: lxydiy <1452206376@qq.com>
Date: Mon, 14 Jul 2025 13:06:15 +0800
Subject: [PATCH] gstplayer: fix gst compilation error on rockchip socs.

---
 src/plugins/gstreamer_video_player/frame.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/plugins/gstreamer_video_player/frame.c b/src/plugins/gstreamer_video_player/frame.c
index 66498ce..65f5a86 100644
--- a/src/plugins/gstreamer_video_player/frame.c
+++ b/src/plugins/gstreamer_video_player/frame.c
@@ -515,6 +515,7 @@ static bool get_plane_sizes_from_video_info(const GstVideoInfo *info, size_t pla
 static bool calculate_plane_size(const GstVideoInfo *info, int plane_index, size_t *plane_size_out) {
     // Taken from: https://github.com/GStreamer/gstreamer/blob/621604aa3e4caa8db27637f63fa55fac2f7721e5/subprojects/gst-plugins-base/gst-libs/gst/video/video-info.c#L1278-L1301
 
+    #ifdef GST_VIDEO_FORMAT_INFO_TILE_SIZE
     #if THIS_GSTREAMER_VER >= GSTREAMER_VER(1, 21, 3)
     if (GST_VIDEO_FORMAT_INFO_IS_TILED(info->finfo)) {
         guint x_tiles = GST_VIDEO_TILE_X_TILES(info->stride[plane_index]);
@@ -523,6 +524,7 @@ static bool calculate_plane_size(const GstVideoInfo *info, int plane_index, size
         return true;
     }
     #endif
+    #endif
 
     gint comp[GST_VIDEO_MAX_COMPONENTS];
     guint plane_height;
-- 
2.34.1

